cilium/cmd/service_update.go

```diff
@@ -97,12 +106,14 @@ func updateService(cmd *cobra.Command, args []string) {
 		spec.Flags = &models.ServiceSpecFlags{}
 	}
 
-	if k8sExternalIPs && k8sNodePort {
-		Fatalf("Cannot set both --k8s-external and --k8s-node-port for a service")
+	if boolToInt(k8sExternalIPs)+boolToInt(k8sNodePort)+boolToInt(k8sHostPort) > 1 {
+		Fatalf("Cannot set more than one of --k8s-external, --k8s-node-port, --k8s-host-port for a service")
```

Comment:
Nit, maybe rephrase this to:

```suggestion
		Fatalf("Can only set one of --k8s-external, --k8s-node-port, --k8s-host-port for a service")
```